### PR TITLE
fix: build

### PR DESCRIPTION
**Pull request by Helias**

### CODE REVIEW

File: src/BetterItemReloading.cpp [hunk 1]
```diff
@@ -73,23 +72,23 @@ class BetterItemReloading : public CommandScript
                 "startquest, lockid, Material, sheath, RandomProperty, RandomSuffix, block, itemset, MaxDurability, area, Map, BagFamily, "
                 "TotemCategory, socketColor_1, socketContent_1, socketColor_2, socketContent_2, socketColor_3, socketContent_3, socketBonus, "
                 "GemProperties, RequiredDisenchantSkill, ArmorDamageModifier, duration, ItemLimitCategory, HolidayId, ScriptName, DisenchantID, "
-                "FoodType, minMoneyLoot, maxMoneyLoot, flagsCustom FROM item_template WHERE entry = %u", entry);
+                "FoodType, minMoneyLoot, maxMoneyLoot, flagsCustom FROM item_template WHERE entry = %u", *entry);
```
Comment (Winfidonarleyan) [inline src/BetterItemReloading.cpp:76]: `prifntf` format style not used in actual core

Comment (Helias) [reply src/BetterItemReloading.cpp:76]: ? let me know how to do it properly and how to fix the ChatCommandBuilder warning
apparently, there is something wrong with this but I did not understand how to fix it, I checked mod-transmog and it seems identical

```C++
/home/runner/work/mod-better-item-reloading/mod-better-item-reloading/modules/mod-better-item-reloading/src/BetterItemReloading.cpp:20:11: fatal error: 'ChatCommandBuilder' is deprecated: char const* parameters to command handlers are deprecated; convert this to a typed argument handler instead [-Wdeprecated-declarations]
          { "item",  HandleBetterReloadItemsCommand,  SEC_ADMINISTRATOR, Acore::ChatCommands::Console::Yes }
          ^
/home/runner/work/mod-better-item-reloading/mod-better-item-reloading/src/server/game/Chat/ChatCommands/ChatCommand.h:252:11: note: 'ChatCommandBuilder' has been explicitly marked deprecated here
        [[deprecated("char const* parameters to command handlers are deprecated; convert this to a typed argument handler instead")]]
          ^
1 error generated.
make[2]: *** [modules/CMakeFiles/modules.dir/build.make:104: modules/CMakeFiles/modules.dir/mod-better-item-reloading/src/BetterItemReloading.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1199: modules/CMakeFiles/modules.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

https://github.com/azerothcore/mod-better-item-reloading/actions/runs/3659655497/jobs/6185901653

Comment (Winfidonarleyan) [reply src/BetterItemReloading.cpp:76]: example ChatCommand - https://github.com/azerothcore/mod-cfbg/blob/master/src/cs_cfbg.cpp

Comment (Helias) [reply src/BetterItemReloading.cpp:76]: fixed, thank you, the issue was in the string type of the handler function

now I have this and I don't know what is the issue, may it's related to the `printnf` that you mentioned before?

```
In file included from /Users/4009093/projects/azerothcore-wotlk/deps/fmt/include/fmt/format.h:48:
/Users/4009093/projects/azerothcore-wotlk/deps/fmt/include/fmt/core.h:1709:3: fatal error: static_assert failed due to requirement 'formattable' "Cannot format an argument. To make type T formattable provide a formatter<T> specialization: https://fmt.dev/latest/api.html#udt"
  static_assert(
  ^
/Users/4009093/projects/azerothcore-wotlk/deps/fmt/include/fmt/core.h:1835:23: note: in instantiation of function template specialization 'fmt::detail::make_arg<true, fmt::basic_format_context<fmt::appender, char>, fmt::detail::type::custom_type, std::optional<unsigned int> &, 0>' requested here
        data_{detail::make_arg<
                      ^
/Users/4009093/projects/azerothcore-wotlk/deps/fmt/include/fmt/core.h:1854:10: note: in instantiation of function template specialization 'fmt::format_arg_store<fmt::basic_format_context<fmt::appender, char>, std::optional<unsigned int>, unsigned int, ItemFlagsExtra>::format_arg_store<std::optional<unsigned int> &, unsigned int &, ItemFlagsExtra &>' requested here
  return {std::forward<Args>(args)...};
         ^
/Users/4009093/projects/azerothcore-wotlk/deps/fmt/include/fmt/core.h:3101:28: note: in instantiation of function template specialization 'fmt::make_format_args<fmt::basic_format_context<fmt::appender, char>, std::optional<unsigned int> &, unsigned int &, ItemFlagsExtra &>' requested here
  return vformat(fmt, fmt::make_format_args(args...));
                           ^
/Users/4009093/projects/azerothcore-wotlk/modules/mod-better-item-reloading/src/BetterItemReloading.cpp:238:25: note: in instantiation of function template specialization 'fmt::format<std::optional<unsigned int> &, unsigned int &, ItemFlagsExtra>' requested here
                        LOG_ERROR("sql.sql", "Item (Entry: {}) has value ({}) in `AllowableRace` races, not compatible with ITEM_FLAGS_EXTRA_HORDE_ONLY ({}) in Flags field, item cannot be equipped or used by these races.",
                        ^
/Users/4009093/projects/azerothcore-wotlk/src/common/Logging/Log.h:160:5: note: expanded from macro 'LOG_ERROR'
    LOG_MESSAGE_BODY(filterType__, LogLevel::LOG_LEVEL_ERROR, __VA_ARGS__)
    ^
/Users/4009093/projects/azerothcore-wotlk/src/common/Logging/Log.h:150:17: note: expanded from macro 'LOG_MESSAGE_BODY'
                LOG_EXCEPTION_FREE(filterType__, level__, __VA_ARGS__); \
                ^
/Users/4009093/projects/azerothcore-wotlk/src/common/Logging/Log.h:134:58: note: expanded from macro 'LOG_EXCEPTION_FREE'
            sLog->outMessage(filterType__, level__, fmt::format(__VA_ARGS__)); \
                                                         ^
1 error generated.
make[2]: *** [modules/CMakeFiles/modules.dir/mod-better-item-reloading/src/BetterItemReloading.cpp.o] Error 1
make[1]: *** [modules/CMakeFiles/modules.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 98%] Built target scripts
make: *** [all] Error 2
```

Comment (Helias) [reply src/BetterItemReloading.cpp:76]: full build here https://github.com/azerothcore/mod-better-item-reloading/actions/runs/3712294646/jobs/6293669929

Comment (Helias) [reply src/BetterItemReloading.cpp:76]: @Winfidonarleyan ?

Comment (Winfidonarleyan) [reply src/BetterItemReloading.cpp:76]: need replace `printf` fmt to `fmt`. For this need replace `%u` to `{}`

**Comment by EricksOliveira** [comment]
I tried to compile.

Erro	C2338	static_assert failed: 'Cannot format an argument. To make type T formattable provide a formatter<T> specialization: https://fmt.dev/latest/api.html#udt'	modules	C:\Server\Arena-BR\azerothcore-wotlk\deps\fmt\include\fmt\core.h	1710

**Comment by pangolp** [comment]
@Helias Shall we close this PR? Between today and tomorrow, I'll start updating the module.

**Pull request closed without merging**